### PR TITLE
Add quit launcher after game stops option (Steam Deck)

### PR DESCRIPTION
**Pull request by d-513**
Allows for launcher to quit directly after the game stops. This is useful for the Deck where you want to just get to the steam menu
 Best combined with "Hide launcher after game window starts"

### CODE REVIEW

**Review by txtsd — APPROVED** [review]

**Review by ZekeZDev — APPROVED** [review]

File: launcher/minecraft/launch/LauncherPartLaunch.cpp [hunk 1]
```diff
@@ -168,8 +168,14 @@ void LauncherPartLaunch::on_state(LoggedProcess::State state)
         }
         case LoggedProcess::Finished:
         {
+            if(APPLICATION->settings()->get("QuitAfterGameStop").toBool())
+            {
+                APPLICATION->closeAllWindows();
+                APPLICATION->exit();
+            }
```
Comment (Scrumplex) [inline launcher/minecraft/launch/LauncherPartLaunch.cpp]: I don't like that this is here. LauncherPartLaunch is one of many steps of a game launch. If we close here, all later steps will probably not run anymore (or there will be a race condition). This should be implemented as an additional step (see https://github.com/PolyMC/PolyMC/blob/92f3154e8f7c1421541a00ae7cf57f188cfe46a5/launcher/minecraft/MinecraftInstance.cpp#L906-L933)

**Review by Scrumplex — CHANGES_REQUESTED** [review]

File: launcher/minecraft/launch/LauncherPartLaunch.cpp (hunk 1 above)
Comment (d-513) [reply launcher/minecraft/launch/LauncherPartLaunch.cpp]: I agree, will try to fix when I'm home

File: launcher/launch/steps/QuitAfterGameStop.cpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include "QuitAfterGameStop.h"
+#include <launch/LaunchTask.h>
+#include "Application.h"
+
+void QuitAfterGameStop::executeTask()
+{
+    APPLICATION->quit();
+}
```
Comment (Scrumplex) [inline launcher/launch/steps/QuitAfterGameStop.cpp]: nit: no newline at the end of the file

File: launcher/launch/steps/QuitAfterGameStop.cpp [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#include "QuitAfterGameStop.h"
```
Comment (Scrumplex) [inline launcher/launch/steps/QuitAfterGameStop.cpp:19]: nit: license header maybe?

File: launcher/launch/steps/QuitAfterGameStop.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (Scrumplex) [inline launcher/launch/steps/QuitAfterGameStop.h:19]: nit: license header maybe?

**Review by Scrumplex — APPROVED** [review]
looks good

**Comment by Scrumplex** [comment]
You can use this as the license header for new files:
```c
// SPDX-License-Identifier: GPL-3.0-only
/*
 *  PolyMC - Minecraft Launcher
 *  Copyright (C) 2022 dada513 <dada513@protonmail.com>
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, version 3.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

File: launcher/launch/steps/QuitAfterGameStop.cpp (hunk 2 above)
Comment (d-513) [reply launcher/launch/steps/QuitAfterGameStop.cpp]: ok

**Pull request merged**